src/Grpc.Net.Client/Balancer/Internal/BalancerHttpHandler.cs

```diff
@@ -46,13 +46,17 @@ public BalancerHttpHandler(HttpMessageHandler innerHandler, HttpHandlerType http
                 var socketsHttpHandler = HttpRequestHelpers.GetHttpHandlerType<SocketsHttpHandler>(innerHandler);
                 CompatibilityHelpers.Assert(socketsHttpHandler != null, "Should have handler with this handler type.");
 
-                socketsHttpHandler.ConnectCallback = OnConnect;
+                if (!HttpRequestHelpers.IsSocketsHttpHandlerSetup(socketsHttpHandler))
```

Comment:
Is thread safety a concern? I followed this back to GrpcChannel.ForAddress etc. which I'm guessing is meant to be setup once on a single thread.